src/coffee/helpers/task-queue.coffee

```diff
@@ -49,13 +50,19 @@ class TaskQueue
     @_activeCount += 1
 
     task.fn()
-    .then (res) ->
-      task.resolve res
-    .catch (error) ->
-      task.reject error
+    .then (res) =>
+      # postpone resolving the task
+      @_completedTasks.push(task.resolve.bind(this, res))
+    .catch (error) =>
+      # postpone rejecting the task
+      @_completedTasks.push(task.reject.bind(this, error))
```

Comment:
i think we should reject/resolve here ASAP otherwise all task callers have to wait until all promises are resolved?